ansible_rulebook/job_template_runner.py

```diff
@@ -91,6 +80,35 @@ def _sslcontext(self) -> Union[bool, ssl.SSLContext]:
                 return ssl.create_default_context(cafile=self.verify_ssl)
         return False
 
+    async def _get_job_template_id(self, name: str, organization: str) -> int:
+        slug = f"{self.JOB_TEMPLATE_SLUG}/"
+        params = {"search": name}
+
+        async with aiohttp.ClientSession(
+            headers=self._auth_headers()
+        ) as session:
+            while True:
+                response = await self._get_page(session, slug, params)
+                json_body = json.loads(response["body"])
+                for jt in json_body["results"]:
```

Comment:
I believe you are assuming that the request went fine and that could not happen. Should it check if the request returned 200 and `results` is in the response? otherwise you could have unexpected "jsondecode or keyerror" exceptions if there are issues with the awx connection, don't?

Reply:
ok, I see the status code is checked in the `_get_page` method. Should it check if the response is deserializable and it contains the `results` key?

Reply:
Controller list API always returns results as an array. I think we can trust the return has the desired format.